spring-kafka-docs/src/main/antora/modules/ROOT/pages/kafka/annotation-error-handling.adoc

```diff
@@ -242,6 +242,14 @@ public DefaultErrorHandler errorHandler(ConsumerRecordRecoverer recoverer) {
 }
 ----
 
+[IMPORTANT]
+The `DefaultErrorHandler` only processes exceptions that inherit from `RuntimeException`.
+Exceptions inheriting from `Error` bypass the error handler entirely, causing the consumer to terminate immediately, close the Kafka connection, and skip all retry/recovery mechanisms.
+This critical distinction means applications may report healthy status despite having terminated consumers that no longer process messages.
+Always ensure that exceptions thrown in message processing code explicitly extend from `RuntimeException` rather than `Error` to allow proper error handling.
+In other words, if the application throws an exception, ensure that it is extended from `RuntimeException` and not inadvertently inherited from `Error`.
+Standard errors like `OutOfMemoryError`, `IllegalAccessError`, and other errors beyond the control of the application are still treated as `Error`s and not retried.
```

Comment:
There was a recent fix for similar plural words: https://github.com/spring-projects/spring-kafka/commit/4272118e35d4b10f6cd8e9da60efa09bb7562d3b